### DISCUSSION TITLE
fabric-api doesn't automatically download

### DISCUSSION THREAD

**Discussion by Alddste** [itzg/docker-minecraft-server#2539] [Q&A]
I'm assuming this isn't an actual issue, which is why I am asking this question here. I decided to create a docker-compose file which keeps my server updated. However, the fabric-api doesn't download despite it being labelled properly. I have checked and the fabric-api is for the latest version (1.20.4). Any help would be appreciated. Thanks!

```version: "3.8"
services:
  mc:
    image: itzg/minecraft-server
    environment:
      UID: "1001"
      GID: "1001"
      TZ: America/Denver
      EULA: "true"
      TYPE: "fabric"
      VERSION: "latest"
      FABRIC_LOADER_VERSION: "latest"
      MEMORY: 64G
      SEED: 5599410981053532367
      OPS: "yeet"
      DIFFICULTY: "hard"
      MAX_PLAYERS: "1487"
      VIEW_DISTANCE: "11"
      SIMULATION_DISTANCE: "11"
      ENABLE_WHITELIST: "true"
      ENFORCE_WHITELIST: "true"
      OVERRIDE_WHITELIST: "true"
      ENFORCE_SECURE_PROFILE: "false"
      WHITELIST: "yeet"
      MOTD: "yeet"
      SPAWN_PROTECTION: "-1"
      MODRINTH_DOWNLOAD_DEPENDENCIES: "required"
      MODRINTH_PROJECTS: "alternate-current,anti-xray,debugify,c2me-fabric:alpha,fabric-api,incendium,ksyxis,lithium,modernfix,no-chat-reports,nullscape,servercore,server-side-replay-recorder,spark,starlight,styled-chat,styled-nicknames,styledplayerlist,terralith,vmp-fabric:alpha"
      DATAPACKS: /datapacks
      SYNC_CHUNK_WRITES: "false"
      NETWORK_COMPRESSION_THRESHOLD: "64"
      RCON_CMDS_STARTUP:  |-
        gamerule doFireTick false
        gamerule disableElytraMovementCheck true
        gamerule doImmediateRespawn true
        gamerule showDeathMessages true
        gamerule playersSleepingPercentage 0
      ENABLE_AUTOPAUSE: "true"
      MAX_TICK_TIME: "-1"
    ports:
      - "25565:25565"
    volumes:
      - ./data:/data
      - ./mods:/mods
      - ./datapacks:/datapacks
      - ./config:/config
    restart: unless-stopped
volumes:
  data: {}
```

**Comment by itzg** [comment]
Your compose file looks right as far as I can tell. Can you try a couple of things?

To make sure you have the latest image in case something got fixed, run `docker compose pull`. See if fabric-api downloads after that. 

If not still, set the environment variable `DEBUG` to "true" and provide the container startup logs (ideally just the logs before the Minecraft server itself starts).

**Reply by Alddste** [reply]
I ran `docker compose pull` and the server seems to crash. Here is the output.

```[+] Running 1/0
 ✔ Container mc-mc-1  Created                                                                                  0.0s
Attaching to mc-mc-1
mc-mc-1  | [init] Running as uid=1001 gid=1001 with /data as 'drwxrwxr-x 1 1001 1001 1032 Dec 11 18:23 /data'
mc-mc-1  | [init] Autopause functionality enabled
mc-mc-1  | [init] Starting RCON commands
mc-mc-1  | [init] Rcon cmds functionality enabled
mc-mc-1  | [init] Resolving type given fabric
mc-mc-1  | [mc-image-helper] 18:27:17.701 INFO  : Fabric launcher for minecraft 1.20.4 loader 0.15.1 is already available
mc-mc-1  | [init] Copying datapacks from /datapacks ...
mc-mc-1  | [mc-image-helper] 18:27:19.364 INFO  : Downloading styledplayerlist-3.3.0+1.20.3.jar
mc-mc-1  | [mc-image-helper] 18:27:19.469 ERROR : Invalid parameter provided for 'modrinth' command: No files are available for the project i7rbL330 for loader fabric and Minecraft version 1.20.4
mc-mc-1 exited with code 0```

**Reply by itzg** [reply]
```
Invalid parameter provided for 'modrinth' command: No files are available for the project i7rbL330 for loader fabric and Minecraft version 1.20.4
```

I'm guessing you didn't notice Minecraft have upgraded to 1.20.4 and you'll need to set VERSION to 1.20.2 probably.

**Reply by Alddste** [reply]
I do see that, my apologies for not including a question. Will the boot up not bypass mods that aren't completely up to date? The mod which is out of date is called [thecolonel63's Server Side Replay Recorder](https://modrinth.com/mod/server-side-replay-recorder). I was thinking that the docker boot process would simply not download any jars which aren't compatible and boot, allowing. I'm assuming that if this is not available due to stability reasons, it could be done with a sidecar?

**Reply by itzg** [reply]
No since that will leave the installation in an inconsistent state with non-obvious error reports and server failures. Those types of failures are quite hard to support and I'd rather avoid them in the first place.
